### ISSUE TITLE
Cannot find module 'vue-router' or its corresponding type declarations. v0.14.2

### ISSUE THREAD

**Issue by Handpear** [hannoeru/vite-plugin-pages#63]
use vite-plugin-pages in react project

npm run build ("build": "tsc && vite build --mode production")

node_modules/vite-plugin-pages/client.d.ts:3:34 - error TS2307: Cannot find module 'vue-router' or its corresponding type declarations.

3   import { RouteRecordRaw } from 'vue-router'
                                   ~~~~~~~~~~~~

node_modules/vite-plugin-pages/client.d.ts:10:34 - error TS2307: Cannot find module 'vue-router' or its corresponding type declarations.

10   import { RouteRecordRaw } from 'vue-router'
                                    ~~~~~~~~~~~~

node_modules/vite-plugin-pages/dist/index.d.ts:2:41 - error TS2307: Cannot find module 'vue-router' or its corresponding type declarations.

2 import { RouteLocationNormalized } from 'vue-router';
                                          ~~~~~~~~~~~~

package.json ↓

{
    "author": "handpear",
    "name": "vite-react-antd-concent-typescript-windicss-alloy-admin",
    "version": "0.0.0",
    "scripts": {
        "dev": "vite --mode development",
        "test": "vite --mode test",
        "build": "tsc && vite build --mode production",
        "preview": "vite preview"
    },
    "dependencies": {
        "antd": "^4.16.5",
        "concent": "^2.15.12",
        "crypto-js": "^4.0.0",
        "nprogress": "^0.2.0",
        "qs": "^6.10.1",
        "react": "^17.0.2",
        "react-dom": "^17.0.2",
        "react-router": "^5.2.0",
        "react-router-config": "^5.1.1",
        "react-router-dom": "^5.2.0",
        "umi-request": "^1.3.9"
    },
    "devDependencies": {
        "@types/crypto-js": "^4.0.1",
        "@types/node": "^15.12.5",
        "@types/nprogress": "^0.2.0",
        "@types/qs": "^6.9.6",
        "@types/react": "^17.0.11",
        "@types/react-dom": "^17.0.7",
        "@types/react-router-config": "^5.0.2",
        "@types/react-router-dom": "^5.1.7",
        "@typescript-eslint/eslint-plugin": "^4.27.0",
        "@typescript-eslint/parser": "^4.27.0",
        "@vitejs/plugin-react-refresh": "^1.3.4",
        "@vue/compiler-sfc": "^3.1.2",
        "autoprefixer": "^10.2.6",
        "concent-plugin-redux-devtool": "^2.0.8",
        "eslint": "^7.28.0",
        "eslint-config-alloy": "^4.1.0",
        "eslint-plugin-react": "^7.24.0",
        "less": "^4.1.1",
        "postcss-assets": "^6.0.0",
        "typescript": "^4.3.3",
        "vite": "^2.3.8",
        "vite-plugin-components": "^0.11.2",
        "vite-plugin-html": "^2.0.7",
        "vite-plugin-pages": "^0.14.2",
        "vite-plugin-style-import": "^1.0.1",
        "vite-plugin-windicss": "^1.0.4",
        "vite-react-jsx": "^1.1.1",
        "windicss": "^3.1.3"
    }
}
should i install vue-router?

**Comment by hannoeru** [comment]
Nope, just type check error, that file is only used in Vue, I will remove it.

**Comment by Handpear** [comment]
There will still be an error in 0.14.3

**Comment by Handpear** [comment]
node_modules/vite-plugin-pages/dist/index.d.ts:2:41 - error TS2307: Cannot find module 'vue-router' or its corresponding type declarations.

2 import { RouteLocationNormalized } from 'vue-router';、

Found 1 error.

**Comment by hannoeru** [comment]
Can you show me your `tsconfig.json`?

**Comment by Handpear** [comment]
// tsconfig.json
{
    "compilerOptions": {
        "target": "ESNext",
        "module": "ESNext",
        "lib": ["DOM", "DOM.Iterable", "ESNext"],
        "allowJs": true,
        "checkJs": true,
        "jsx": "react",
        "declaration": true,
        "sourceMap": true,
        "removeComments": true,
        "noEmit": true,
        "importHelpers": true,
        "isolatedModules": true,

        "strict": true,
        "noImplicitAny": true,
        "strictNullChecks": true,
        "noImplicitThis": true,
        "alwaysStrict": true,
        "forceConsistentCasingInFileNames": true,
        "skipLibCheck": false,
        "suppressImplicitAnyIndexErrors": false,

        "noUnusedLocals": true,
        "noUnusedParameters": false,
        "noImplicitReturns": true,
        "noFallthroughCasesInSwitch": true,
        "noImplicitOverride": true,

        "resolveJsonModule": false,
        "esModuleInterop": true,
        "moduleResolution": "Node",
        "baseUrl": "./",
        "paths": {
            "@/*": ["src/*"],
            "public": ["public"],
            "configs": ["configs"]
        },
        "typeRoots": ["node_modules/@types", "src/types"],
        "types": ["vite/client"],
        "allowSyntheticDefaultImports": true,

        "experimentalDecorators": true,
        "emitDecoratorMetadata": true
    },
    "include": ["src/**/*", "configs/**/*", "typings/**/*"],
    "exclude": ["node_modules", "build", "dist", "scripts", "jest"]
}

**Comment by Handpear** [comment]
// vite-env.d.ts

/// <reference types="vite/client" />
/// <reference types="vite-plugin-pages/client-react" />

**Comment by Handpear** [comment]
/// reference types="vite/client" /
/// reference types="vite-plugin-pages/client-react" /

**Comment by hannoeru** [comment]
Set `emitDecoratorMetadata ` to `false` and try it again?

**Comment by Handpear** [comment]
I tried, but it didn't work
After I upgrade to the new version, it's still the same

**Comment by hannoeru** [comment]
Can you provide a repro?

**Comment by Handpear** [comment]
https://gitee.com/handpear/vite-react-antd-concent-typescript-windicss-alloy-admin.git

of course, this is my repository address

**Comment by hannoeru** [comment]
Fixed in v0.15.0
